enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,411 @@
+
+###### `--kustomize` (new project layouts)
+
+Generate a `kustomize` base CSV for a new project layout under `config/packages/bases`.
+This base will be used as an actual base for `kustomize build` eventually; for now
+only `generate packagemanifests` will read this base. This step involves generating UI
+metadata from API types, although `--manifests` can do this as well. This subcommand
+is interactive if a base does not exist.
+
+New project layout defaults:
+* Output directory: `config/packages/bases`
+* Read from: `--apis-dir="api(s)"`
+
+###### `--manifests`
+
+Generate a package containing a CSV and owned CRD manifests, and updates the
+[package manifest file][package-manifest-doc]. This step uses a base generated
+with `--kustomize`, or creates a new one. This subcommand is interactive if a
+base does not exist.
+
+New project layout defaults:
+* Output directory: `config/packages/<version>`
+* Read from: stdin, `--apis-dir="api(s)"`
+
+Old project layout defaults:
+* Output directory: `deploy/olm-catalog/<operator-name>/<version>`
+* Read from: `--manifest-root="deploy"`, `--crds-dir="deploy/crds"`, `--apis-dir="pkg/apis"`
+
+Notable flags: `--version=<semver>`, `--channel=<channel-name>`, `--default-channel=<channel-name>`
+
+#### `make` recipes (new project layouts)
+
+New project layouts are scaffolded with a [Makefile][kb-makefile] containing a
+set of pre-defined recipes for generating manifests, building images, etc. To align
+with this idiom, recipes should be amended/added to include `generate bundle` by
+default. If a user does not want these features turned on in their project, they
+can manually remove the amendments/additions.
+
+###### Modifying the base scaffold
+
+`operator-sdk init` will scaffold a Makefile containing the following amendments/additions:
+
+```make
+# Current Operator version
+VERSION ?= 0.0.1
+# Default bundle image tag
+BUNDLE_IMG ?= controller-bundle:$(VERSION)
+# Options for 'bundle-build'
+ifeq ($(origin BUNDLE_CHANNELS), undefined)
+BUNDLE_CHANNELS = --channels=$(BUNDLE_CHANNELS)
+endif
+ifeq ($(origin BUNDLE_DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(BUNDLE_DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+...
+
+# Existing manifests recipe, append 'generate bundle'
+manifests:
+  ...
+  operator-sdk generate bundle --kustomize
+
+...
+
+# Generate bundle manifests
+bundle: manifests
+	kustomize build config/bundle | operator-sdk generate bundle --manifests --version $(VERSION)
+
+# Generate bundle manifests with a particular version and metadata,
+# validate generated bundle files, and build the bundle image.
+bundle-build: manifests
+	kustomize build config/bundle | operator-sdk generate bundle --manifests --metadata --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
+	operator-sdk bundle validate config/bundle
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+```
```

Comment:
Just to clarifies, the operator-framework/operator-sdk#2860 is updating the following target which is NOT documented here and is what I understand that is affecting the default behaviour and the backwards compatibility with kubebuilder.

```
# Generate manifests e.g. CRD, RBAC etc.
manifests: controller-gen
	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
	# operator-sdk generate bundle -q --kustomize // this line was added 
```

Reply:
It is documented [here](https://github.com/operator-framework/enhancements/pull/16/files#diff-b4a68ba3ae57e617574dfe5811fb6ae7R260).